apps/prairielearn/src/models/course.ts

```diff
@@ -12,3 +24,51 @@ export async function selectCourseById(course_id: string): Promise<Course> {
     CourseSchema,
   );
 }
+
+export async function selectCoursesWithStaffAccess({
+  course_id,
+  user_id,
+  is_administrator,
+  authz_data_overrides,
+}: {
+  /** The ID of the current course. This is used when authz overrides are in place. */
+  course_id: string;
+  user_id: string;
+  is_administrator: boolean;
+  authz_data_overrides: any | null | undefined;
+}) {
+  const { courses } = await callValidatedOneRow(
+    'courses_with_staff_access',
+    [user_id, is_administrator],
+    z.object({
+      courses: z.array(CourseWithPermissionsSchema),
+    }),
+  );
+
+  // If any overrides are in place, we'll confine the user to the current course.
+  if (authz_data_overrides) {
+    return courses.filter((c) => idsEqual(c.id, course_id));
+  }
```

Comment:
Now that this was removed from the switcher, there are only two calls to this function (both indirect through `selectCoursesWithEditAccess`) that pass a non-null value to this parameter:

* in a template course, copying a template question to your own course;
* in question settings, duplicating a question to a different course.

Neither of these seems to be a case where showing authorized courses other than the current one would be an issue when effective user/date/mode/etc. is in use. Is it still worth doing this filter?

Reply:
I think there's a question of exactly how much fidelity (for lack of a better word) we want effective user overrides to have. That is: if I impersonate user X, we'd have a higher-fidelity experience as X if we don't see courses that X wouldn't have access to.

Is this fidelity worth the complexity here? I think I'm leaning towards no, and removing `course_id` and `authz_data_overrides` from this function.

Reply:
I agree with "no" here. To make sure I'm saying "no" to the right thing, I mean that even when emulating another user we should still be able to copy questions to courses that the emulated user can't access.

Reply:
I made this change in 786752f. I also added some comments to remind our future selves that no filtering is done if effective user overrides are in place. It should be obvious from the arguments and implementation, but better safe than sorry, right?